src/sentry/utils/versioning.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+import warnings
+from collections import namedtuple
+
+from sentry.exceptions import InvalidConfiguration
+
+
+class Version(namedtuple('Version', 'major minor patch')):
+    def __str__(self):
+        return '.'.join(map(str, self))
+
+
+def make_upgrade_message(service, modality, version, hosts):
+    return '{service} {modality} be upgraded to {version} on {hosts}.'.format(
+        hosts=', '.join('{0} (currently {1})'.format(*i) for i in hosts),
```

Comment:
This host list might be really long. One per shard.

Not sure if you'd want to truncate it and say, "host1, host2, host3, and 5 others" or something.